### ISSUE TITLE
low map

### ISSUE THREAD

**Issue by asdfqwer2015** [ammirato/target_driven_instance_detection#6]
Hi:
  I trained a model with configAVD2, after training(by default, the model will be saved at the 15th epoch end), the model reaches only 4.19% map. which is too low.

  I set almost all the parameters to default, except some params like dataset dir etc. I modified resize params like https://github.com/ammirato/target_driven_instance_detection/issues/1#issuecomment-379806941

  I also modified one line of codes due to nan in different pytorch version:
 https://github.com/ammirato/target_driven_instance_detection/issues/2#issuecomment-479760899
  But the loss can normally decrease without nan, so I think this modification is correct.

  And the precisions are about 0.04, the recalls are about 0.18. I missed the eval details.
  And here are some outputs from test_tdid.py, there are too many false alarms.

![out_img3](https://user-images.githubusercontent.com/26498687/55856454-f120df80-5b9c-11e9-914e-a00cb8d8f4ce.jpg)
![out_img2](https://user-images.githubusercontent.com/26498687/55856456-f120df80-5b9c-11e9-8654-cddd58308ff5.jpg)
![out_img1](https://user-images.githubusercontent.com/26498687/55856458-f120df80-5b9c-11e9-863c-1925c075bd4a.jpg)

  I noticed that predicted 300 anchor scores for each test img are all in range about 0.52~0.57. So I think the model may failed to train with cross_entropy_loss. And x,y seems to have appropriate variance, but w and h seems to be consistently.  

  BTW, the avg_epoch_loss of trainset of this evaluated model is 3.18 at 15th epoch end, and batchsize is 5.

  What did I missed?
  Thanks.

**Comment by ammirato** [comment]
What pytorch version are you using? If you are not using pytorch 0.3.X, then I do not expect the code to work.

What is your inital learning rate? It should .001 as mentioned in the paper, not .0001 as in the current config file. The config file should be updated.

Also you should try training for 50 epochs with the learning rate at .001, and then another 15 epochs with the learning rate at .0001.

**Comment by asdfqwer2015** [comment]
Thanks for your quick reply.
My pytorch version is 1.0.1.
And I retrained a model with SAVE_FREQ=1, and got a increased map trend, from 0.00226 at epoch1 to 0.13926 at epoch16. The model looks have not trained sufficient. And there might be some thing wrong at my prev training(0.0419 at epoch15).
And I'll try the new lr schedular since the training has a increase trend.

**Comment by asdfqwer2015** [comment]
Once I have the new results, I'll post here.

**Comment by asdfqwer2015** [comment]
En, I got map 41.2 map after 50epcoh(lr0.001) + 15epoch(lr 0.0001). Is it normally?
for the step2(15 epoch with lr 0.0001), I trained based on the eval best model(33th epoch) from step1. Should I use latest model instead of best model from step1?

**Comment by ammirato** [comment]
I would suggest trying both, but if your pytorch version is not 0.3.X, there could be other problems reducing the performance. Currently this code is only supported for pytorch 0.3.X.
